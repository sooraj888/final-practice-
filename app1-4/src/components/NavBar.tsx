import { AppBar, Toolbar, Typography } from "@material-ui/core";
import React from "react";

const NavBar = (): JSX.Element => {
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography>Country Weather App</Typography>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
