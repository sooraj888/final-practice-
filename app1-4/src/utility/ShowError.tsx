import { Box } from "@material-ui/core";
import React from "react";

const ShowError = (): JSX.Element => {
  return (
    <Box>
      <h1>
        <div>Oops</div>
        Somthing Went wrong!
      </h1>
    </Box>
  );
};

export default ShowError;
