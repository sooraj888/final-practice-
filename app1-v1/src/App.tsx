import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { Box } from "@material-ui/core";

function App() {
  return <Box className="App"></Box>;
}

export default App;
