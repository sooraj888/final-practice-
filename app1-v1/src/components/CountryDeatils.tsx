import {
  Box,
  Button,
  Card,
  CardContent,
  CardMedia,
  colors,
  Typography,
} from "@material-ui/core";

import { makeStyles } from "@material-ui/styles";
import React, { useEffect, useState } from "react";
import { Navigate, NavigateFunction, useNavigate } from "react-router-dom";
import useFeacthApi from "../hooks/useFeacthApi";

const useStyles = makeStyles({
  root: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    margin: "30px",
  },
  root1: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "cenetr",
  },
  title: {
    fontWeight: 700,
    color: "gray",
  },
  subTitle: { fontSize: "20px", fontWeight: 900 },
  root3: {
    display: "flex",
    padding: "30px",
  },
  btn: {
    width: "100%",
  },
});

const CountryDeatils = ({ countryData }: any): JSX.Element => {
  const classes = useStyles();
  const navigate: NavigateFunction = useNavigate();
  const [showWeather, setShowWeather] = useState<boolean>(false);
  const URL: string =
    "http://api.weatherstack.com/current?access_key=78ae6ba411469f63795fe8974e0e3a45&query=";
  const { data } = useFeacthApi(URL, countryData?.capital?.[0]);

  useEffect((): void => {
    console.log(data);
  }, [data]);

  console.log(countryData);
  return (
    <Box className={classes.root}>
      {!showWeather ? (
        <Card className={classes.root1}>
          <CardMedia
            component={"img"}
            image={countryData?.flags?.png}
            alt="flag"
          ></CardMedia>
          <CardContent>
            <Typography className={classes.title}>County</Typography>
            <Typography className={classes.subTitle}>
              {countryData?.name?.common}
            </Typography>
            <Typography className={classes.title}>Capital</Typography>
            <Typography className={classes.subTitle}>
              {countryData?.capital?.[0]}
            </Typography>
            <Typography className={classes.title}>Population</Typography>
            <Typography className={classes.subTitle}>
              {countryData?.population}
            </Typography>
            <Typography className={classes.title}>Lat-Lng</Typography>
            <Typography className={classes.subTitle}>
              {countryData?.latlng?.[0]}-{countryData?.latlng?.[1]}
            </Typography>
          </CardContent>
          <Button
            variant="contained"
            color="primary"
            className={classes.btn}
            onClick={() => {
              setShowWeather(true);
            }}
          >
            Capital Weather
          </Button>
        </Card>
      ) : (
        <Card>
          <Box className={classes.root3}>
            <CardContent>
              <Typography className={classes.title}>Region</Typography>
              <Typography className={classes.subTitle}>
                {data?.location?.region}
              </Typography>
              <Typography className={classes.title}>Temperature</Typography>
              <Typography className={classes.subTitle}>
                {data?.current?.temperature}
              </Typography>
              <Typography className={classes.title}>Wind-Speed</Typography>
              <Typography className={classes.subTitle}>
                {data?.current?.wind_speed}
              </Typography>
            </CardContent>
            <Box>
              <CardMedia
                component={"img"}
                image={data?.current?.weather_icons?.[0]}
                alt="weather icon"
              ></CardMedia>
              <CardContent>
                <Typography>{data?.current?.weather_descriptions}</Typography>
              </CardContent>
            </Box>
          </Box>

          <Button
            color="primary"
            className={classes.btn}
            variant="contained"
            onClick={(): void => {
              navigate("/");
            }}
          >
            Search Other Country
          </Button>
        </Card>
      )}
    </Box>
  );
};

export default CountryDeatils;
