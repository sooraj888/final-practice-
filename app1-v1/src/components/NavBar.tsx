import { AppBar, Toolbar, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import React from "react";

const useStyles = makeStyles({
  roots: {
    fontSize: "40px",
  },
});

const NavBar = (): JSX.Element => {
  const classes = useStyles();
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography className={classes.roots}>Country Weather App</Typography>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
