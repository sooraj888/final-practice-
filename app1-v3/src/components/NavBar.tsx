import { AppBar, Toolbar, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import React from "react";

const useStyle = makeStyles({
  root: {
    fontSize: "40px",
  },
});

const NavBar = (): JSX.Element => {
  const classes = useStyle();
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography className={classes.root}>
          Country and Weather App
        </Typography>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
