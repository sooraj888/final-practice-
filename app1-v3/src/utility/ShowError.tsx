import React from "react";

const ShowError = (): JSX.Element => {
  return <div>Error</div>;
};

export default ShowError;
