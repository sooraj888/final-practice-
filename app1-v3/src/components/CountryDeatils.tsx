import {
  Box,
  Button,
  Card,
  CardContent,
  CardMedia,
  Typography,
} from "@material-ui/core";
import React, { useState } from "react";
import { makeStyles } from "@material-ui/styles";
import { NavigateFunction, useNavigate } from "react-router-dom";
import useAPIFeatch, { useAPIFeatchType } from "../hooks/useAPIFeatch";
import ShowError from "../utility/ShowError";
import { classicNameResolver } from "typescript";

const useStyle = makeStyles({
  root: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  root2: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    marginTop: "30px",
    width: "200px",
    height: "400px",
    border: "1px solid black",
    justifyContent: "space-around",
    padding: "20px",
    borderRadius: "10px",
  },
});

const CountryDeatils = ({ countryData }: any): JSX.Element => {
  const classes = useStyle();
  const [showWeather, setShowWeather] = useState<boolean>(false);
  const navigate: NavigateFunction = useNavigate();

  const url: string =
    "http://api.weatherstack.com/current?access_key=78ae6ba411469f63795fe8974e0e3a45&query=";

  const { data }: useAPIFeatchType = useAPIFeatch(url, countryData?.capital);

  //   console.log(countryData);

  console.log(data);
  return (
    <Box
      sx={{ display: "flex", flexDirection: "column", alignItems: "center" }}
    >
      {!showWeather ? (
        <Card className={classes.root}>
          <CardMedia
            component={"img"}
            image={countryData?.flags?.png}
            alt="flag"
          ></CardMedia>
          <CardContent>
            <Typography>Country Name</Typography>
            <Typography>{countryData?.name?.common}</Typography>
            <Typography> Capital</Typography>
            <Typography>{countryData?.capital}</Typography>
            <Typography> Population</Typography>
            <Typography>{countryData?.population}</Typography>
            <Typography> Lat-Lng</Typography>
            <Typography>
              {countryData?.latlng?.[0]}-{countryData?.latlng?.[0]}
            </Typography>
          </CardContent>
          <Button
            onClick={(): void => {
              setShowWeather(true);
            }}
            variant="contained"
            color="primary"
          >
            Weather Deatails
          </Button>
        </Card>
      ) : (
        <Card>
          {!data ? (
            <ShowError></ShowError>
          ) : (
            <>
              <Box>
                <CardContent>
                  <Typography>Region</Typography>
                  <Typography>{data?.location?.region}</Typography>
                  <Typography>Temperature</Typography>
                  <Typography>{data?.current?.temperature}</Typography>
                  <Typography>Wind Speed</Typography>
                  <Typography>{data?.current?.wind_speed}</Typography>
                </CardContent>
                <Box>
                  <CardMedia
                    image={data?.current?.weather_icons[0]}
                    alt="weather icon"
                    component={"img"}
                  ></CardMedia>
                  <Typography>{data?.current?.weather_descriptions}</Typography>
                </Box>
              </Box>
            </>
          )}
          <Button
            onClick={(): void => {
              navigate("/");
            }}
          >
            Search Another Country
          </Button>
        </Card>
      )}
    </Box>
  );
};

export default CountryDeatils;
