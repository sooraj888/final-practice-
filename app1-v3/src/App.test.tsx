import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";
import NavBar from "./components/NavBar";
import ShowError from "./utility/ShowError";
import CountryDeatils from "./components/CountryDeatils";
import { BrowserRouter } from "react-router-dom";

describe("test on NavBar Componet", () => {
  test("rendering NavBar ", () => {
    render(<NavBar />);
    screen.debug();
  });
});

describe("test on NavBar Componet", () => {
  test("finding text inside NavBar ", () => {
    render(<NavBar />);
    const element = screen.getByText(/Country and Weather App/);
    expect(element).toBeInTheDocument();
  });
});

describe("test on NavBar Componet", () => {
  test("finding text inside NavBar ", () => {
    const { container } = render(<NavBar />);

    expect(container).toMatchSnapshot();
  });
});

describe("test on NavBar Componet", () => {
  test("finding text inside NavBar ", () => {
    render(<NavBar />);
    const element = screen.getByText(/Country and Weather App/);
    expect(element).toHaveAttribute("class");
  });
});

describe("test on ShowError Componet", () => {
  test("rendering ShowError ", () => {
    render(<ShowError />);
    screen.debug();
  });
});

describe("test on ShowError Componet", () => {
  test("finding text inside ShowError ", () => {
    render(<ShowError />);
    const element = screen.getByText(/Error/);
    expect(element).toBeInTheDocument();
  });
});

describe("test on ShowError Componet", () => {
  test("snapshot testing on ShowError ", () => {
    const { container } = render(<ShowError />);

    expect(container).toMatchSnapshot();
  });
});

describe("test on ShowError Componet", () => {
  test("finding attribute of particular element", () => {
    render(<ShowError />);
    const element = screen.getByText(/Error/);
    expect(element).not.toHaveAttribute("class");
  });
});

const data = {
  name: { common: "india" },
  capital: "dehli",
  population: "130cr",
  flags: { png: "" },
};

describe("test on CountryDeatils Componet", () => {
  test("rendering CountryDeatils without prop ", () => {
    render(
      <BrowserRouter>
        <CountryDeatils countryData={""} />
      </BrowserRouter>
    );
    screen.debug();
  });
});

describe("test on CountryDeatils Componet", () => {
  test("rendering CountryDeatils with prop ", () => {
    render(
      <BrowserRouter>
        <CountryDeatils countryData={data} />
      </BrowserRouter>
    );
    screen.debug();
  });
});

describe("test on CountryDeatils Componet", () => {
  test("finding text inside CountryDeatils which is sent as prop ", () => {
    render(
      <BrowserRouter>
        <CountryDeatils countryData={data} />
      </BrowserRouter>
    );
    const element = screen.getByText(/Lat-Lng/);
    expect(element).toBeInTheDocument();

    expect(screen.getByText(/130cr/)).toBeInTheDocument();
    expect(screen.getByText(/dehli/)).toBeInTheDocument();
  });
});

describe("test on CountryDeatils Componet", () => {
  test("snapshot testing on CountryDeatils ", () => {
    const { container } = render(
      <BrowserRouter>
        <CountryDeatils countryData={data} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});

describe("test on CountryDeatils Componet", () => {
  test("finding attribute of particular element", () => {
    render(
      <BrowserRouter>
        <CountryDeatils countryData={data} />
      </BrowserRouter>
    );
    screen.debug();
    const element = screen.getByText(/india/);
    expect(element).toHaveAttribute("class");
  });
});

//  //intensional error this will throw erro because after clicking button the text was removed so we get erro  here "screen.getByText(/130cr/)"
describe("test on CountryDeatils Componet", () => {
  test("intensional error this will throw erro because after clicking button the text was removed so we get erro  here ", () => {
    const { container } = render(
      <BrowserRouter>
        <CountryDeatils countryData={data} />
      </BrowserRouter>
    );
    screen.debug();
    expect(screen.getByText(/130cr/)).toBeInTheDocument();

    fireEvent.click(screen.getByRole("button"));

    // expect(screen.getByText(/130cr/)).toBeInTheDocument();
  });
});
