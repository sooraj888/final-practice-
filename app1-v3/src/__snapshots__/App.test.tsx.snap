// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on CountryDeatils Componet fire event test  1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <img
        alt="flag"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      />
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Country Name
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          india
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
           Capital
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          dehli
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
           Population
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          130cr
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
           Lat-Lng
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          -
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Weather Deatails
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`test on CountryDeatils Componet fire event test  2`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div>
        Error
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Search Another Country
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`test on CountryDeatils Componet snapshot testing on CountryDeatils  1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <img
        alt="flag"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      />
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Country Name
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          india
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
           Capital
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          dehli
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
           Population
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          130cr
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
           Lat-Lng
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          -
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Weather Deatails
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`test on NavBar Componet finding text inside NavBar  1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Country and Weather App
      </p>
    </div>
  </header>
</div>
`;

exports[`test on ShowError Componet snapshot testing on ShowError  1`] = `
<div>
  <div>
    Error
  </div>
</div>
`;
